docs/spec/escrow.rst

```diff
@@ -0,0 +1,312 @@
+Parameter       Type          Description                                            
+==============  ============  ========================================================
+``amount``      Balance       The amount deposited to receive vote-escrowed tokens.
+``end``         BlockNumber   The end height after which the balance can be unlocked.
+==============  ============  ========================================================
+
+Point
+.....
+
+The ``bias``, ``slope`` and ``ts`` for our linear function.
+
+.. tabularcolumns:: |l|L|
+
+==============  ============  ========================================================	
+Parameter       Type          Description                                            
+==============  ============  ========================================================
+``bias``        Balance       The bias for the linear function.
+``slope``       Balance       The slope for the linear function.
+``ts``          BlockNumber   The current block height when this point was stored.
+==============  ============  ========================================================
+
+Functions
+~~~~~~~~~
+
+.. _escrow-function-create-lock:
+
+create_lock
+-----------
+
+Create a lock on the account's balance to expire in the future.
+
+Specification
+.............
+
+*Function Signature*
+
+``create_lock(who, amount, unlock_height)``
+
+*Parameters*
+
+* ``who``: The user's address.
+* ``amount``: The amount to be locked.
+* ``unlock_height``: The height to lock until.
+
+*Events*
+
+* :ref:`escrow-event-deposit`
+
+*Preconditions*
+
+* The function call MUST be signed by ``who``.
+* The ``amount`` MUST be non-zero.
+* The account ``who`` MUST NOT already have locked balance.
+* The ``unlock_height`` MUST be greater than ``now``.
+* The ``unlock_height`` MUST NOT be greater than ``now + MaxPeriod``.
+
+*Postconditions*
+
+* The account's ``LockedBalance`` MUST be set as follows:
+
+    * ``new_locked.amount``: MUST be the ``amount``.
+    * ``new_locked.end``: MUST be the ``unlock_height``.
+
+.. _escrow-function-increase-amount:
+
+increase_amount
+---------------
+
+Deposit additional tokens for a pre-existing lock to improve voting power.
+
+Specification
+.............
+
+*Function Signature*
+
+``increase_amount(who, amount)``
+
+*Parameters*
+
+* ``who``: The user's address.
+* ``amount``: The amount to be locked.
+
+*Events*
+
+* :ref:`escrow-event-deposit`
+
+*Preconditions*
+
+* The function call MUST be signed by ``who``.
+* The ``amount`` MUST be non-zero.
+* The account's ``old_locked.amount`` MUST be non-zero.
+* The account's ``old_locked.end`` MUST be greater than ``now``.
+
+*Postconditions*
+
+* The account's ``LockedBalance`` MUST be set as follows:
+
+    * ``new_locked.amount``: MUST be ``old_locked.amount + amount``.
+    * ``new_locked.end``: MUST be the ``old_locked.end``.
+
+.. _escrow-function-increase-unlock-height:
+
+increase_unlock_height
```

Comment:
`extend_unlock_height`?

Reply:
Based on the naming used by [curve](https://github.com/curvefi/curve-dao-contracts/blob/702c9b59de1542d78656c9c949e9ee8e6d206543/contracts/VotingEscrow.vy#L450), but happy to change.